analyzers/src/SonarAnalyzer.Common/Rules/UseUnixEpochBase.cs

```diff
@@ -0,0 +1,116 @@
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), "DateTimeOffset"));
+                    }
+                },
+                Language.SyntaxKind.ObjectCreationExpressions);
+        });
+
+    protected static bool IsValueEqualsTo(TLiteralExpression literal, int value) =>
+        literal.ChildTokens().First().ValueText == value.ToString();
+
+    private bool IsEpochCtor(SyntaxNode node, SemanticModel model)
+    {
+        var methodSymbol = (IMethodSymbol)model.GetSymbolInfo(node).Symbol;
+        var lookup = Language.MethodParameterLookup(node, methodSymbol);
+
+        if (IsParameterExistingAndLiteralEqualTo("year", 1970, lookup)
+            && IsParameterExistingAndLiteralEqualTo("month", 1, lookup)
+            && IsParameterExistingAndLiteralEqualTo("day", 1, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("hour", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("minute", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("second", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("millisecond", 0, lookup)
+            && IsParameterNonExistingOrLiteralEqualTo("microsecond", 0, lookup)
+            && IsDateTimeKindNonExistingOrUtc(lookup)
+            && IsCalendarNonExistingOrGregorian(lookup)
+            && IsOffsetNonExistingOrZero(lookup))
+        {
+            return true;
+        }
+        return false;
+    }
+
+    private static bool IsParameterExistingAndLiteralEqualTo(string parameterName, int value, IMethodParameterLookup lookup) =>
+        lookup.TryGetSyntax(parameterName, out var expressions) && IsLiteralAndEqualTo(expressions[0], value);
+
+    private static bool IsParameterNonExistingOrLiteralEqualTo(string parameterName, int value, IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax(parameterName, out var expressions) || IsLiteralAndEqualTo(expressions[0], value);
+
+    private static bool IsLiteralAndEqualTo(SyntaxNode node, int value) =>
+        node is TLiteralExpression literal && IsValueEqualsTo(literal, value);
+
+    private bool IsDateTimeKindNonExistingOrUtc(IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax("kind", out var expressions)
+        || (expressions[0] is TMemberAccessExpression memberAccess && IsDateTimeKindUtc(memberAccess));
+
+    private bool IsCalendarNonExistingOrGregorian(IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax("calendar", out var expressions) || IsGregorianCalendar(expressions[0]);
+
+    private bool IsOffsetNonExistingOrZero(IMethodParameterLookup lookup) =>
+        !lookup.TryGetSyntax("offset", out var expressions) || IsZeroTimeOffset(expressions[0]);
+
+    // DateTime.UnixEpoch introduced at .NET Core 2.1/.NET Standard 2.1
+    private static bool CompilationTargetsValidNetVersion(Compilation compilation) =>
```

Comment:
How about `IsUnixEpochProprtySupported`?

Reply:
I think I prefer `CompilationTargetsValidNetVersion`. It's also consistent with a few other examples in the repo

Reply:
Furthermore, should this be cached?

Reply:
IMO this way of checking the framework will be replaced in the near future with a formalized way of doing it (and consistent with all the rules). Having the same temporary, yet unprecise name might help to spot the affected rules and transition to the new system. Having said that, I can update the name as you suggest since mine is not a strong opinion👍